src/tls.c

```diff
@@ -910,11 +914,17 @@ static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
 
     if (conn->c.state != CONN_STATE_CONNECTED) return -1;
     ERR_clear_error();
+    if (data_len < conn->last_failed_write_data_len) {
+        conn->c.last_errno = EBUSY;
+        return -1;
+    }
```

Comment:
lets place a comment explaining this logic and why we think it is O.K to do so.

Reply:
@ranshid I think we should also try to re-introduce the debugServerAssert. We can do it at the calling sites though, https://github.com/valkey-io/valkey/blob/cf19deb61d8ea0ba01a61f57edc1e9440367d98b/src/networking.c#L2255 I suppose might be a good place to check. Not a blocker for me on this PR.

Reply:
@madolson placing a debugAssert would still cause the tests to fail right?

Reply:
also I am not sure about your proposal. In the current suggestion the error is not modifying the connection state (which is indeed strange) I am starting to think that even modifying the conn->c.last_errno is not needed since we would just want to return -1.

Reply:
I think we should have this
```suggestion
    /* In case when last write failed due to some internal reason, retry has to provide
     * at least the same amount of bytes (https://docs.openssl.org/master/man3/SSL_write).
     * If that condition is not met, OpenSSL will return "SSL routines::bad length".
     * When this assertion gets hit, caller has to be fixed to provide required bytes. */
    debugServerAssert(data_len >= conn->last_failed_write_data_len);
```

Default test config has `enable-debug-assert yes` and this should be enough during tests(do I miss anything?), but prevent Valkey crash on production, in worst case it should lead to previous behaviour `SSL routines::bad length`, that should mean to us that some other places in code are not providing the same buffer size (or greater).

Let me know is it looking good for you

Reply:
Yeah, we want this to be a debug assert since it will crash during tests BUT will just result in disconnects in production workloads. This looks right to me.